### ISSUE TITLE
DataCloneError in postMessage

### ISSUE THREAD

**Issue by OEvgeny** [vuejs/vue-devtools#18]
I found that for some reason there are a couple of components in my app witch trigger this error:
```
Uncaught DataCloneError: Failed to execute 'postMessage' on 'Window': An object could not be cloned.
```
It is in [this](https://github.com/vuejs/vue-devtools/blob/014eab0cfdcf04200b95b444315a854a9f2e5c25/shells/chrome/src/backend.js#L23-L29) piece of code..

Can we do something with this?

**Comment by yyx990803** [comment]
Can you add a debugger statement there and see what `data` contains? This error is usually thrown when the data to be sent contains non-plain objects e.g. functions.

**Comment by OEvgeny** [comment]
Yes I did it. It contains component props. One of them was a function.

**Comment by OEvgeny** [comment]
```js
event: "instance-details"
payload: Object
    id: 4
    name: "Datatable"
    state: Array[5]
        0: Object
            key: "items"
            meta: Object
            type: "prop"
            value: Array[8]
            __proto__: Object
        1: Object
            key: "onSelect"
            meta: Object
            binding mode: "default"
            required: false
            type: "Function"
            __proto__: Object
...
```

**Comment by yyx990803** [comment]
Can you be more specific about where the function was found?

**Comment by OEvgeny** [comment]
Did you mean component definition? If yes, it looks like:
```js
<script>
module.exports: {
  methods: {
    doSomething () {/* */}
  },
  components: {
    datatable: {
      props: {
        items: /*  */,
        onSelect {
          type: Function,
          default () {return function () {}}
        }
      }
    }
  }
}
</script>

<template>
<datatable :on-select="doSomething"></datatable>
</template>
```

When I click to `<datatable>` in vue-devtools I am getting error described above.
It may be because in [shells/chrome/src/backend.js line 26](https://github.com/vuejs/vue-devtools/blob/014eab0cfdcf04200b95b444315a854a9f2e5c25/shells/chrome/src/backend.js#L26) `data.payload.state[1].type` is a `Function` for this component.

If something is wrong with my answer, sorry :) Let me know if I can provide additional information for this.

**Comment by OEvgeny** [comment]
It looks like I did something the other way this time. There is `value` property in `data.payload.state[1]` what contains function... So it looks like:
```js
event: "instance-details"
...
payload: Object
  state: [5]
    ...
    1: Object
      key: "onSelect"
      meta: Object
      type: "prop"
      value: function (a)
      __proto__: Object
```

**Comment by OEvgeny** [comment]
Adding filtration for cases when value is function makes it work well:
```js
  if (data.payload.state) {
    data.payload.state = data.payload.state.map(function (el) {
      return !(el.value instanceof Function)
    })
  }
```
It seems we should somehow replace these functions with something else...

**Comment by yyx990803** [comment]
Ok, should be fixed in 1.0.1 with proper sanitization before sending the data.

**Comment by OEvgeny** [comment]
Thanks, Evan!
